nni/compression/tensorflow/compressor.py

```diff
@@ -233,6 +312,10 @@ def __init__(self, layer, config, pruner):
         _logger.info('Layer detected to compress: %s', self.layer.name)
 
     def call(self, *inputs):
+        self._update_weights()
+        return self.layer(*inputs)
+
+    def _update_weights(self):
```

Comment:
why we split this function?

Reply:
Because forward should not be the only allowed place to update weights.